src/vsg/app/Window.cpp

```diff
@@ -41,7 +41,7 @@ Window::Window(ref_ptr<WindowTraits> traits) :
 {
     if (_traits && (_traits->swapchainPreferences.surfaceFormat.format == VK_FORMAT_B8G8R8A8_SRGB || _traits->swapchainPreferences.surfaceFormat.format == VK_FORMAT_B8G8R8_SRGB))
     {
-        _clearColor = linear_to_sRGB(_clearColor);
+        _clearColor = sRGB_to_linear(_clearColor);
```

Comment:
If the swapchain is SRGB then why would one need to convert from sRGB_to_linear?

The default _clearColor has been linear up to this point.

Reply:
The image format was previously `_UNORM` with the colour space set to sRGB, so there was no automatic from-sRGB conversion happening during the clear before this PR, and the value of `_clearColor` was what would end up on an sRGB screen. As `linear_to_sRGB` was really doing a `sRGB_to_linear` conversion before this PR, that meant you'd end up with a nonsense colour, but it looked fine because it's not like the precise maths to derive the clear colour particularly matters. When the framebuffer image format was changed to `_SRGB`, it made Vulkan start doing an automatic linear-to-sRGB conversion during the clear, which combined with the fix to make the colour space conversion functions match their names, made the clear colour really dark due to a double linear-to-sRGB conversion. As people typically type out colours in perceptually uniform sRGB space (e.g. they'd chosen it with a colour picker in an image editor, which nearly always use sRGB by default), and Vulkan was now expecting a linear colour, it seemed most reasonable to do a sRGB-to-linear conversion here.

Reply:
I am now inclined to merge this PR as a branch and start testing.  I think the changes to Window.cpp really need explanation as what is going on why it looks like something perverse is being done.

Reply:
The original function usage didn't make sense. The original `_clearColor` value was being interpreted as sRGB due to the `_UNORM` image format paired with `_SRGB_NONLINEAR` colour space, and yet the `{0.2f, 0.2f, 0.4f, 1.0f}` was going through an operation claiming to be an sRGB-to-linear conversion. Obviously, you don't convert something from the colour space it's about to be interpreted as to a different one.

If `{0.2f, 0.2f, 0.4f, 1.0f}` is linear, then the correct original behaviour would be a linear-to-sRGB conversion (which is what the function call said it was doing, but it was instead doing the opposite) and the new correct behaviour would be to do no conversion at all.

If `{0.2f, 0.2f, 0.4f, 1.0f}` is sRGB, then the correct original behaviour would be to do no conversion at all, and the new correct behaviour would be to do an sRGB-to-linear conversion.

If the displayed colour on the screen was correct, then that would suggest that `{0.2f, 0.2f, 0.4f, 1.0f}` was in a made-up colour space with a 4.84 gamma value (as it became sRGB when undergoing an sRGB-to-linear conversion), and the sensible thing to do would be to write it out in a real colour space (whether sRGB or linear) instead.

As it seemed unlikely that you'd knowingly have used a mad colour space, called `sRGB_to_linear` on a colour you thought was already linear, or ignored a clear colour that was wildly different to the one you thought you'd specified, it seemed most plausible that `{0.2f, 0.2f, 0.4f, 1.0f}` was sRGB, so the code now does what's most appropriate for an sRGB colour.

Reply:
Okay, it seems *something's* slightly off in my explanation. I made a test image:

![image](https://github.com/user-attachments/assets/bf85a19d-2b90-4434-8f9c-ba0ef7344998)

Both with and without this PR, the clear colour is displayed like the middle row on my machine. That makes sense with this PR, but as it was previously doing a sRGB-to-linear conversion (mislabeled as linear-to-sRGB) and then supposedly no further conversions, it doesn't make sense without this PR. I'm digging further to see what's going on.

Reply:
Well that was dumb. I'd somehow missed that the conversion was happening in an `if` block predicated on the swapchain format. I'll take another look and revise some of what I said.

Reply:
Thanks for trying to come up with coherent explanation.

The clearColor values used are simply what worked for the OSG and I wanted to keep the background consistent to avoid too much of jump for users going between the OSG and VSG.  The implicit assumption was that the clear colour was in linear RGB colour space, the frame buffer and final display would also be in linear RGB colour space.

From my reading of what you've written is the assumption that the final display would honour the linear RGB colour space of the frame buffer is misplaced, and that monitors would be treating the linear RGB frame buffer as straight input without conversion and would be displaying it in sRGB, but... these monitors are assuming the frame buffers are already in sRGB rather linear RGB, so the whole chain of conflicting assumptions is what we are essence trying to untangle.